File: rh-front/src/app/models/stagiaire.ts
import { User } from "./user";

export class Stagiaire {
    id: number;
    civilite: string;
    niveau_etudes: string;
    ville: string;
    telephone: string;
    cvData : any;
    LinkedIn_URL: string;
    userId: number;
    user: User;

    constructor() {
        this.id = 0;
        this.civilite = '';
        this.niveau_etudes = '';
        this.ville = '';
        this.telephone = '';
        // this.cv = '';
        this.LinkedIn_URL = '';
        this.userId = 0;
        this.user = new User();
    }
}

File: stagiaire/src/main/java/com/rh/stagiaire/Entities/Stagiaire.java
package com.rh.stagiaire.Entities;

import com.rh.stagiaire.Model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Data @NoArgsConstructor @AllArgsConstructor @ToString
public class Stagiaire {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String civilite;
    private String niveau_etudes;
    private String ville;
    private String telephone;
    private String cv;
    private String LinkedIn_URL;
    private Long userId;
    @Transient
    private User user;


}
